movedata/move.go

```diff
@@ -82,14 +90,24 @@ func (mover *Mover) MoveData() error {
 			return err
 		}
 
+		convertedArrayOfData := casting.CastArrayOfData(columnsMap, cols, columns)
+		for i := range columns {
+			columnPointers[i] = &convertedArrayOfData[i]
+		}
+
 		document := make(map[string]interface{})
 		for i, colName := range cols {
 			val := columnPointers[i].(*interface{})
 			document[colName] = *val
 		}
 
 		for _, jsonfield := range mover.queryConf.JSONFields {
+			//TODO: check if jsonField is a json itself
+
+			//TODO: Build the json structure
```

Comment:
yep, here you will need to cycle every key of the json parsed map, in a very similar way of .CastArrayOfData

Reply:
Done